examples/evaluator/golang/simple/evaluate/evaluator_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package evaluate
+
+import (
+	"testing"
+
+	structpb "github.com/golang/protobuf/ptypes/struct"
+	"github.com/stretchr/testify/assert"
+	"open-match.dev/open-match/examples"
+	harness "open-match.dev/open-match/pkg/harness/evaluator/golang"
+	"open-match.dev/open-match/pkg/pb"
+)
+
+func TestEvaluate(t *testing.T) {
+	ticket1 := &pb.Ticket{Id: "1"}
+	ticket2 := &pb.Ticket{Id: "2"}
+	ticket3 := &pb.Ticket{Id: "3"}
+
+	ticket12Score1 := &pb.Match{
+		Ticket: []*pb.Ticket{ticket1, ticket2},
+		Properties: &structpb.Struct{
+			Fields: map[string]*structpb.Value{
+				examples.MatchScore: {Kind: &structpb.Value_NumberValue{NumberValue: 1}},
+			},
+		},
+	}
+
+	ticket12Score10 := &pb.Match{
+		Ticket: []*pb.Ticket{ticket2, ticket1},
+		Properties: &structpb.Struct{
+			Fields: map[string]*structpb.Value{
+				examples.MatchScore: {Kind: &structpb.Value_NumberValue{NumberValue: 10}},
+			},
+		},
+	}
+
+	ticket123Score5 := &pb.Match{
+		Ticket: []*pb.Ticket{ticket1, ticket2, ticket3},
+		Properties: &structpb.Struct{
+			Fields: map[string]*structpb.Value{
+				examples.MatchScore: {Kind: &structpb.Value_NumberValue{NumberValue: 5}},
+			},
+		},
+	}
+
+	tests := []struct {
+		description string
+		testMatches []*pb.Match
+		wantMatches []*pb.Match
+	}{
+		{
+			description: "test deduplicates and expect the one with higher score",
+			testMatches: []*pb.Match{ticket12Score1, ticket12Score10},
+			wantMatches: []*pb.Match{ticket12Score10},
+		},
+		{
+			description: "test first returns matches with higher score",
+			testMatches: []*pb.Match{ticket123Score5, ticket12Score10},
+			wantMatches: []*pb.Match{ticket12Score10, ticket123Score5},
```

Comment:
Here the wantMatches is just ticket12Score10. The Evaluator has to de-dupe and pick only one match that uses a ticket. The goal of deduping should be - any ticket that has appeared in a 'result' should not reappear in across (hence we add it to ignore list to even sit it out of future rounds)